src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -47,8 +47,15 @@ public void init() {
    * properly.
    */
   public void prepareMessageForDisplay(Message message) {
-    String regex = "(https?://\\S+\\.(png|jpg))";
-    String replacement = "<img src=\"$1\" />";
+    /**
```

Comment:
I think it's pretty good, but I have two comments:
1) "Javadoc" style comments `/**  */` are for documenting methods and classes, but regular comments should use `//`.
2) I think you might be able to simplify slightly by giving a real example, and a little bit of prose description.  Maybe something like:

```
// Matches the URL of an image file, with an optional caption. For example:
//     [the google logo] http://www.google.com/images/logo.png
// Matched URLs must end with one of: .png, .jpg, .gif
String regex = "(\\[.+\\])?\\s(https?://[\\S+\\.]+/([\\S+\\.?]+/?)+\\.(png|jpg|gif))";

// Replaces the URL with the actual image. If a caption was given, it is printed below the image.
String replacement = "<figure><img src=\"$2\" /> <figcaption>$1</figcaption></figure>";
```